### PR TITLE
Add auth hash to external Id

### PR DESCRIPTION
**Pull request by Jeasmine**
* Add auth to setExternalId
* Add tests for new functionality

Needs name params validation

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-android-sdk/1210)

### CODE REVIEW

**Review by jkasten2 — APPROVED** [review]

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java [hunk 1]
```diff
@@ -1565,8 +1595,20 @@ public void run() {
                return;
             }
 
+            if (remoteParams != null && remoteParams.useEmailAuth && externalIdAuthHash == null) {
```
Comment (jkasten2) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1598]: `remoteParams.useEmailAuth` should be ` remoteParams.useUserIdAuth`

**Review by jkasten2 — CHANGES_REQUESTED** [review]
Need to add `require_user_id_auth` to remote params

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java [hunk 2]
```diff
@@ -1565,8 +1595,20 @@ public void run() {
...
+               String errorMessage = "External Id authentication (auth token) is set to REQUIRED for this application. Please provide an auth token from your backend server or change the setting in the OneSignal dashboard.";
+               if (completionCallback != null)
+                  completionCallback.onFailure(new ExternalIdError(ExternalIdErrorType.REQUIRES_EXTERNAL_ID_AUTH, errorMessage));
+               Log(LOG_LEVEL.ERROR, errorMessage);
+               return;
+            }
+
+            String internalExternalIdAuthHash = externalIdAuthHash;
```
Comment (emawby) [inline OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java:1606]: Can we change the name of internalExternalIdAuthHash? Is the only change we are making is making it all lower case? If that is the case maybe include the lowercase aspect in the name?

**Review by emawby — CHANGES_REQUESTED** [review]

**Review by jkasten2 — APPROVED** [review]
`external_user_id_auth_hash` is correctly added here so I think it looks good to me.

**Pull request merged**